libs/langchain/langchain/vectorstores/scann.py

```diff
@@ -0,0 +1,544 @@
+            **kwargs,
+        )
+        return [doc for doc, _ in docs_and_scores]
+
+    def similarity_search(
+        self,
+        query: str,
+        k: int = 4,
+        filter: Optional[Dict[str, Any]] = None,
+        fetch_k: int = 20,
+        **kwargs: Any,
+    ) -> List[Document]:
+        """Return docs most similar to query.
+
+        Args:
+            query: Text to look up documents similar to.
+            k: Number of Documents to return. Defaults to 4.
+            filter: (Optional[Dict[str, str]]): Filter by metadata. Defaults to None.
+            fetch_k: (Optional[int]) Number of Documents to fetch before filtering.
+                      Defaults to 20.
+
+        Returns:
+            List of Documents most similar to the query.
+        """
+        docs_and_scores = self.similarity_search_with_score(
+            query, k, filter=filter, fetch_k=fetch_k, **kwargs
+        )
+        return [doc for doc, _ in docs_and_scores]
+
+    @classmethod
+    def __from(
+        cls,
+        texts: List[str],
+        embeddings: List[List[float]],
+        embedding: Embeddings,
+        metadatas: Optional[List[dict]] = None,
+        ids: Optional[List[str]] = None,
+        normalize_L2: bool = False,
+        **kwargs: Any,
+    ) -> ScaNN:
+        scann = dependable_scann_import()
+        distance_strategy = kwargs.get(
+            "distance_strategy", DistanceStrategy.EUCLIDEAN_DISTANCE
+        )
+        scann_config = kwargs.get("scann_config", None)
+
+        vector = np.array(embeddings, dtype=np.float32)
+        if normalize_L2:
+            vector = normalize(vector)
+        if scann_config is not None:
+            index = scann.scann_ops_pybind.create_searcher(vector, scann_config)
+        else:
+            if distance_strategy == DistanceStrategy.MAX_INNER_PRODUCT:
+                index = (
+                    scann.scann_ops_pybind.builder(vector, 1, "dot_product")
+                    .score_brute_force()
+                    .build()
+                )
+            else:
+                # Default to L2, currently other metric types not initialized.
+                index = (
+                    scann.scann_ops_pybind.builder(vector, 1, "squared_l2")
+                    .score_brute_force()
+                    .build()
+                )
+        documents = []
+        if ids is None:
+            ids = [str(uuid.uuid4()) for _ in texts]
+        for i, text in enumerate(texts):
+            metadata = metadatas[i] if metadatas else {}
+            documents.append(Document(page_content=text, metadata=metadata))
+        index_to_id = dict(enumerate(ids))
+
+        if len(index_to_id) != len(documents):
+            raise Exception(
+                f"{len(index_to_id)} ids provided for {len(documents)} documents."
+                " Each document should have an id."
+            )
+
+        docstore = InMemoryDocstore(dict(zip(index_to_id.values(), documents)))
```

Comment:
Hi - I read the other vectorstore implementations on `from_texts` and I don't see them checking if a docstore is passed in. I agree that this would be a good feature to have - but maybe we do it in a separate pull request when we refactor the interface of `from_texts` instead of this one?

Reply:
yeah i think probably overkill